tests/test_api.py

```diff
@@ -1,34 +1,31 @@
 import os
 
-import pytest
-
 import audb
 import audeer
 
 
-def test_available():
+def test_available(repository):
+
+    # Broken database in repo
+    name = 'non-existing-database'
+    path = os.path.join(
+        repository.host,
+        repository.name,
+        name,
+    )
+    path = audeer.mkdir(path)
+    df = audb.available()
+    os.rmdir(path)
+    assert len(df) == 0
 
     # Non existing repo
     name = 'non-existing-repo'
     audb.config.REPOSITORIES = [
         audb.Repository(
             name=name,
-            host=pytest.FILE_SYSTEM_HOST,
-            backend=pytest.BACKEND,
+            host=repository.host,
+            backend=repository.backend,
         )
     ]
     df = audb.available()
```

Comment:
When I print the repositories it searches it contains:

```
[Repository('data-public-local', 'https://artifactory.audeering.com/artifactory', 'artifactory'), Repository('data-public', 'https://audeering.jfrog.io/artifactory', 'artifactory'), Repository('data-private-local', 'https://artifactory.audeering.com/artifactory', 'artifactory'), Repository('data-local', '~/audb-host', 'file-system'), Repository('data-unittests-local', '/tmp/pytest-of-jwagner/pytest-10/test_available0/host', 'file-system')]
```

Reply:
So it seems to add the temporal repository to the default repositories it reads from `~audb.yaml`.

Reply:
Probably happens in line 88 of `test/conftest.py`

Reply:
Yes, good catch.

I can reproduce it with:

```bash
$ python -m pytest tests/test_api.py
```

The problem in the `repository` fixture is that we can only use `+=` to assign the repository, because otherwise it cannot be used together with the `persistent_repository` fixture that also assigns a repository.
I solved the problem now by adding an additional `hide_default_repository` fixture that first unsets all default repos.

Reply:
Another solution would be to change the code of the `repository` fixture to
```python
...
    current_repositories = audb.config.REPOSITORIES
    audb.config.REPOSITORIES = [repository]
    
    yield repository

    audb.config.REPOSITORIES = current_repositories

```

Then we would never have a mixture of the `persistent_repository` and the `repository` inside a test.
Which is what we want to have anyway for all the tests we have at the moment.

There is one issue (https://github.com/audeering/audb/issues/233), for which we need to have two repositories inside the same test, but maybe we can then add another fixture?

Reply:
Thanks for the explanation and good to know you have to use 'tmpdir_factory` with `module` level!

I still have some remarks:

* Maybe rename `hide_default_repository()` to `hide_default_repositories()` as I mentioned before?
* Since we are no longer using `tmp_path_factory` we should be able to simplify `str(tmpdir_factory.mktemp('host'))` to `tmpdir_factory.mktemp('host')`
* To align the two fixtures maybe use `tmpdir_factory` for both
* And I think it's cleaner to reset the repositories in both fixtures to the original repositories, which at the moment do only with `persistent_repository` I think

Reply:
And once the two fixtures are aligned with `scope` being the only difference, we should maybe introduce a helper function we call from both fixtures.

Reply:
You are right, since we call `yield` we cannot easily move it to a helper function I think.

Reply:
I had another look at it. In principle you can do stuff like:

```python
@pytest.fixture
def create_repo(tmpdir_factory):
    # ... create repo
    yield repository
    # ... tear down stuff


@pytest.fixture
def repository(create_repo):
    return create_repo
```

But the problem is we need to have different scopes for the fixtures `repository` and `persistent_repository`, which is not compatible with calling the same `create_repo()` fixture as this will only be compatible with one scope, e.g. function.

---

So I stayed with repeating code, but added a few more docstrings now.